File: leetcode_solutions/0121-best-time-to-buy-and-sell-stock/solution.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minPrice = prices[0];
        int maxProf = 0;
        for (int price: prices){
            if (price < minPrice){
                minPrice = price;
            } else if (price > minPrice){
                maxProf = max(price - minPrice, maxProf);
            }
        }
        return maxProf;
    }
};

File: leetcode_solutions/0121-best-time-to-buy-and-sell-stock/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_profit = 0
        min_element =prices[0]

        for price in prices:
            if min_element > price:
                min_element = price
            else:
                max_profit = max(max_profit, price - min_element)
        return max_profit
